### PR TITLE
Java: Add Gemini ChatCompletion and HuggingFace

### PR DESCRIPTION
**Pull request by milderhc**
### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: java/pom.xml [hunk 1]
```diff
@@ -70,6 +70,7 @@
         <module>aiservices/openai</module>
 
         <module>semantickernel-api</module>
+        <module>aiservices/google</module>
```
Comment (johnoliver) [inline java/pom.xml:73]: think I would move this up to be with the openai one, would also rename from google to gemini

File: java/aiservices/google/pom.xml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        <version>1.0.2-SNAPSHOT</version>
+        <relativePath>../../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>semantickernel-aiservices-google</artifactId>
+    <name>Semantic Kernel Google Services</name>
+    <description>Google services for Semantic Kernel</description>
+
+    <properties>
+        <maven.compiler.source>17</maven.compiler.source>
```
Comment (johnoliver) [inline java/aiservices/google/pom.xml:18]: Is 17 required?

File: java/pom.xml (hunk 1 above)
Comment (milderhc) [reply java/pom.xml:73]: Following .NET naming here.

File: java/aiservices/google/src/main/java/com/microsoft/semantickernel/aiservices/google/chatcompletion/GeminiChatMessageContent.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+    /**
+     * Gets the function calls.
+     *
+     * @return The function calls.
+     */
+    @Nonnull
+    @SuppressFBWarnings("EI_EXPOSE_REP")
+    public List<GeminiFunctionCall> getGeminiFunctionCalls() {
+        return geminiFunctionCalls;
```
Comment (johnoliver) [inline java/aiservices/google/src/main/java/com/microsoft/semantickernel/aiservices/google/chatcompletion/GeminiChatMessageContent.java:62]: can return `Collections.unmodifiableList` and remove the SuppressFBWarnings

**Review by johnoliver — APPROVED** [review]

**Pull request merged**